Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -51,10 +101,54 @@ public static int[] RemoveFirst(int[] array)
         /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
         {
-            // ToDo: implement.
+            var newArray = ProcessArray(array, index);
+            return newArray;
+        }
+
+        private static int[] ProcessArray(int[] array, int index)
+        {
+            var isValid = IsValidCase(array, index);
+
+            if (!isValid)
+            {
+                return array;
+            }
+
+            array = Remove(array, index);
             return array;
         }
 
+        private static bool IsValidCase(int[] array, int index)
+        {
+            if (array == null || index < 0 || index == array.Length || array.Length == 0)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static int[] Remove(int[] array, int index)
+        {
+            var lengthOldArray = array.Length;
```

Comment:
Do you need to use this variable? It's longer than what you had before. Also not really necessary because you do not reuse it anywhere.

Reply:
yep, if its right I use it just under it to calculate the new length of the array I put the data in that should not be removed

Reply:
I meant except for the following line you don't use this variable. So you could replace it with array.Length in the next line and it would probably look better.
But this is taste I guess.

Reply:
oke, changed